tfs/src/main/java/hudson/plugins/tfs/TeamFoundationServerScm.java

```diff
@@ -391,7 +391,7 @@ public boolean pollChanges(AbstractProject hudsonProject, Launcher launcher, Fil
     @Override
     public boolean processWorkspaceBeforeDeletion(AbstractProject<?, ?> project, FilePath workspace, Node node) throws IOException, InterruptedException {
         Run<?,?> lastRun = project.getLastBuild();

```

Comment:
You can explicitly use `AbstractBuild` here, there is no need to cast the type later